crates/core/tedge_agent/src/main.rs

```diff
@@ -1,7 +1,25 @@
+use cap::Cap;
 use clap::Parser;
+use std::alloc;
+use std::time::Duration;
+
+#[global_allocator]
+static ALLOCATOR: Cap<alloc::System> = Cap::new(alloc::System, usize::MAX);
 
 #[tokio::main]
-async fn main() -> Result<(), anyhow::Error> {
+async fn main() -> anyhow::Result<()> {
     let agent_opt = tedge_agent::AgentOpt::parse();
+
+    let log_memory_interval = agent_opt.log_memory_interval;
+    if log_memory_interval > 0 {
+        tokio::spawn(async move {
+            let interval = Duration::from_secs(log_memory_interval);
+            loop {
+                log::info!("Allocated memory: {} Bytes", ALLOCATOR.allocated());
+                tokio::time::sleep(interval).await;
+            }
+        });
+    }
```

Comment:
Oops, I did the fix in the wrong place.

This must be done inside `tedge_agent::run()` and not in the `tedge_main::main()`. Indeed, the latter is only used by the plain executable and not by the sym link pointing to `tedge`.


=> I will fix that tomorrow.

Reply:
Fixed with https://github.com/thin-edge/thin-edge.io/pull/2693/commits/3f0fc8902f33e0549b98e737373b288aadc5379a